static/app/components/lazyRender.tsx

```diff
@@ -0,0 +1,72 @@
+import {useCallback, useRef, useState} from 'react';
+
+function maybeCleanupObserver(
+  observerRef: React.MutableRefObject<IntersectionObserver | null>
+) {
+  if (!observerRef.current) {
+    return;
+  }
+
+  observerRef.current.disconnect();
+  observerRef.current = null;
+}
+
+function supportsIntersectionObserver(): boolean {
+  return 'IntersectionObserver' in window;
+}
+
+const DEFAULT_OPTIONS: IntersectionObserverInit = {
+  root: null,
+  rootMargin: '0px',
+  threshold: 0,
+};
+
+interface LazyRenderProps {
+  children: React.ReactNode;
+  containerHeight?: number;
+  observerOptions?: Partial<IntersectionObserverInit>;
+}
+
+export function LazyRender(props: LazyRenderProps) {
+  // If the browser doesn't support IntersectionObserver, render the children immediately.
+  const [visible, setVisible] = useState<boolean>(!supportsIntersectionObserver());
+  const observerRef = useRef<IntersectionObserver | null>(null);
+
+  const onRefNode = useCallback(
+    (node: HTMLDivElement | null) => {
```

Comment:
maybe note that the react-aria one doesn't currently pass the callback node to use

Reply:
Actually, it seems like react aria doesnt even expose a intersection observer hook, so we cant use it... they only expose the resize observer hook